src/core/config/defaults.js

```diff
@@ -23,7 +23,10 @@ const defaultOptions = Object.freeze({
   displayRequestDuration: false,
   deepLinking: false,
   tryItOutEnabled: false,
-  requestInterceptor: (a) => a,
+  requestInterceptor: (request) => {
+    request.curlOptions = []
```

Comment:
I've noticed that this will add an unnecessary space:
`curl  -X 'GET' \`
instead of:
`curl -X 'GET' \`

Perhaps instead we should define this as `null` and add a check for an array where it's used:
https://github.com/swagger-api/swagger-ui/blob/d5e56e5d75be4edc777fb8432a484e22af690bf1/src/core/plugins/request-snippets/fn.js#L86-L88
For example change it to:
```js
const curlOptions = request.get("curlOptions")
if (Array.isArray(curlOptions)) {
  addWords(...curlOptions)
}
```

Reply:
We don't want empty space. I would suggest to check for empty list. The code should compensate for it. Not sure if at this point the `curlOptions` is immutable structure or POJO, but assuming immutable.

```js
if (List.isList(curlOptions) && !curlOptions.isEmpty()) { 
```

> Perhaps instead we should define this as null and add a check for a list where it's used:

The point is if, somebody uses empty list as the value for curlOptions, the command should still generate without an empty space.